### PR TITLE
feat: Postgresql single server module

### LINKED ISSUE #302: [Module Proposal]: Azure Database for PostgreSQL - Single Server
### Have you checked this module does not already exist in the registry?

Yes

### Why is the module needed?

While we are aware that Postgresql Single Server is currently [on the retirement path](https://learn.microsoft.com/en-us/azure/postgresql/single-server/whats-happening-to-postgresql-single-server), Postgresql Flexible Server currently has no support for Private Endpoints which is a requirement for us.

Therefore, we'll need a Bicep module for this.

### Module path

database/postgresql-single-server

### Describe the module

This module will allow to create [PostgreSQL Single Server ](https://learn.microsoft.com/en-us/azure/postgresql/single-server/overview-single-server) resource.

### PR DESCRIPTION
**Pull request by OmegaVVeapon**
## Description

Closes #302 

## Adding a new module

- [x] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

File: modules/storage/postgresql-single-server/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Postgresql Single Server
```
Comment (shenglol) [inline modules/storage/postgresql-single-server/README.md:1]:
```suggestion
# PostgreSQL Single Server
```

File: modules/storage/postgresql-single-server/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+This module deploys Postgresql Single Server (Microsoft.DBforPostgreSQL/servers) and optionally available integrations.
```
Comment (shenglol) [inline modules/storage/postgresql-single-server/README.md:3]:
```suggestion
This module deploys PostgreSQL Single Server (Microsoft.DBforPostgreSQL/servers) and optionally available integrations.
```

File: modules/storage/postgresql-single-server/README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+## Description
+
+{{ Add detailed description for the module. }}
```
Comment (shenglol) [inline modules/storage/postgresql-single-server/README.md:7]: Can you add some description for the module? Below is an example generated by ChatGPT if you feel like using it:


This Bicep module deploys a PostgreSQL Single Server instance in Azure with configurable options such as backup retention, firewall rules, private endpoints, and SSL enforcement. It also supports role-based access control, storage autogrow, and infrastructure encryption.

File: modules/storage/postgresql-single-server/README.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+### Example 1
+
+Deploy a Postgresql Single Server with minimal parameters
+
+```bicep
+@secure
+param administratorLoginPassword string
+
+module postgresqlSingleServer 'br/public:storage/postgresql-single-server:1.0' = {
```
Comment (shenglol) [inline modules/storage/postgresql-single-server/README.md:55]: The module will have version `1.0.1` once published. Can you update the version number for the examples?

File: modules/storage/postgresql-single-server/README.md (hunk 3 above)
Comment (OmegaVVeapon) [reply modules/storage/postgresql-single-server/README.md:7]: Not a bad example!
Will use that.
Better than what ChatGPT did for me when it asked it for Bicep help, it hallucinated 4 different non-existent ways to do parameter validation 😆

**Review by shenglol — APPROVED** [review]
:shipit:

**Pull request merged**